src/org/graphstream/ui/javafx/util/FxFileSinkImages.java

```diff
@@ -33,56 +33,41 @@
 import javafx.scene.image.WritableImage;
 import org.graphstream.stream.file.FileSinkImages;
 import org.graphstream.ui.javafx.FxGraphRenderer;
-import org.graphstream.ui.view.GraphRenderer;
+import org.graphstream.ui.view.camera.Camera;
 
 import java.awt.image.BufferedImage;
 
-public class FxImageRenderer implements ImageRenderer {
+/**
+ * {@link FileSinkImages} implementation for JavaFX.
+ *
+ * This will start the JavaFX thread, so you may have to exit the platform if you do not have any other JavaFX
+ * application running. Just call `Platform.exit();` at the end of your program.
+ */
+public class FxFileSinkImages extends FileSinkImages {
 	private Canvas canvas;
 	private WritableImage image;
 	private FxGraphRenderer renderer;
 
-	public FxImageRenderer() {
+	public FxFileSinkImages() {
 		this.renderer = new FxGraphRenderer();
+		this.renderer.open(gg, null);
 	}
 

```

Comment:
Add constructors can be useful if i have my setup immediately. Instead of use all set methods.
```java
public FxFileSinkImages() {
	super();
	this.renderer = new FxGraphRenderer();
	this.renderer.open(gg, null);
}

public FxFileSinkImages(OutputType type, Resolution resolution) {
	super(type, resolution, OutputPolicy.NONE);
	this.renderer = new FxGraphRenderer();
	this.renderer.open(gg, null);
}
	
public FxFileSinkImages(OutputType type, Resolution resolution, OutputPolicy outputPolicy) {
	super(type, resolution, outputPolicy);
	this.renderer = new FxGraphRenderer();
	this.renderer.open(gg, null);
}
```

Same for Swing.

Reply:
Yes, it would be more convenient.